### PR TITLE
[PATCH] Improve function test stability

### LINKED ISSUE #362: v2.1.x Chore - The script should handle the case where neither PYTHON environment variable is set nor python3 command is found. Add proper error handling and exit code.
> The script should handle the case where neither PYTHON environment variable is set nor python3 command is found. Add proper error handling and exit code.

_Originally posted by @codecov-ai[bot] in https://github.com/reactive-firewall/multicast/pull/361#discussion_r2043273029_

```bash
if [[ -x "${PYTHON}" ]] ; then
	PY_3_CMD="${PYTHON}"
else
	PY_3_CMD=$(command -v python3);
	if [[ -z "${PY_3_CMD}" ]] ; then
		printf "%s\n" "::error::No Python interpreter found"
		exit 126
	fi
fi ;
```

Edited by @reactive-firewall

### PR DESCRIPTION
**Pull request by reactive-firewall**
# Patch Notes

## Impacted GHI

 - [x] Closes #362

## Changelog

**Changes in file tests/check_integration_coverage:**
  * Improved the test script to handle more edge-cases where `python3` is missing or unset.
  * slight improvement to skip some additional steps after an early failure

---

### CODE REVIEW

**Comment by reactive-firewall** [comment]
@codecov-ai-reviewer review

File: tests/check_integration_coverage [hunk 1]
```diff
@@ -142,21 +142,32 @@ else
...
+fi ;
+# Verify the Python interpreter is actually working
+if [[ ("${EXIT_CODE}" -eq 0) ]] && ! "${PY_3_CMD}" --version >/dev/null 2>&1 ; then
+	printf "%s\n" "::error::Python interpreter at '${PY_3_CMD}' is not working properly" >&2 ;
+	EXIT_CODE=126 ;
 fi ;
 printf "%s\n" "::debug:: Will Use ${PY_3_CMD}"
 # shellcheck disable=SC2086
 COVERAGE_CMD="${PY_3_CMD} -m coverage run -p --source=multicast -m";
 export COVERAGE_CMD
```
[the remark this answers is not in the document]
Comment (reactive-firewall) [reply tests/check_integration_coverage:167]: 🙅🏻 The current implementation is fine.
